src/settings.plugin.js

```diff
@@ -321,26 +327,54 @@ wikiaJWPlayerSettingsPlugin.prototype.getSuitableCaptionsIndex = function(userLa
 		.indexOf(userLang);
 };
 
-wikiaJWPlayerSettingsPlugin.prototype.captionLangMap = {
-	en: 'English',
-	pl: 'Polski',
-	fr: 'Français',
-	de: 'Deutsch',
-	it: 'Italiano',
-	ja: '日本語',
-	pt: 'Português',
-	ru: 'Русский язык',
-	es: 'Español',
-	zh: '中文'
+wikiaJWPlayerSettingsPlugin.prototype.updateCurrentCaptions = function (data) {
+	for (var i = 0; i < this.captionsList.childNodes.length; i++) {
+		this.captionsList.childNodes[i].classList.remove(isActiveClass);
+	}
+
+	this.captionsList.querySelector('[data-track="' + data.track + '"]').classList.add(isActiveClass);
```

Comment:
are we sure querySelector won't return `null`?

Reply:
it's called as a 'captionsChanged' handler